modules/testModule.py

```diff
@@ -1,138 +1,285 @@
+        # add it to the dictionary and update output to the channel
+        if is_test_response(message.clean_content):
+            response_id = cast(int, get_question_id(message))
+            self.log.info(
+                self.class_name,
+                clean_content=message.clean_content,
+                response_id=response_id,
+                is_at_me=self.is_at_me(message),
+            )
+            self.sent_test[response_id]["received_response"] = self.clean_test_prefixes(
+                message, TEST_RESPONSE_PREFIX
+            )
+            return Response(
+                confidence=8,
+                text=test_response_message,
+                why="this was a test",
+            )
+
+        # If Stampy is already running tests and this message is a request
+        # to test himself, ignore it and reply accordingly
+        if self.utils.test_mode:
+            return Response(
+                confidence=9,
+                text=self.TEST_MODE_RESPONSE_MESSAGE,
+                why="Test already running",
+            )
 
-    def is_at_module(self, message):
+        if message.channel.name != "talk-to-stampy":
+            return Response(
+                confidence=10,
+                text="Testing is only allowed in #talk-to-stampy",
+                why=f"{message.author.name} wanted to test me outside of the #talk-to-stampy channel which is prohibited!",
+            )
+
+        if not is_bot_dev(message.author):
+            return Response(
+                confidence=10,
+                text=f"You are not a bot dev, {message.author.name}",
+                why=f"{message.author.name} wanted to test me but they are not a bot dev",
+            )
+
+        # Otherwise, this is a request for Stampy to run integration tests
+        lowercase_module_names = [
+            module_name.lower() for module_name in self.utils.modules_dict
+        ]
+        if match := re.search(r"test module (\w+)", message.clean_content, re.I):
+            parsed_module_name = match.group(1)
+            if parsed_module_name.lower() not in lowercase_module_names:
+                return Response(
+                    confidence=10,
+                    text=f'I don\'t have a module named "{parsed_module_name}"',
+                    why=(
+                        f"{message.author.name} asked me to test module "
+                        f'"{parsed_module_name}" but I don\'t have such a module'
+                    ),
+                )
+            module_name = next(
+                module_name
+                for module_name in self.utils.modules_dict
+                if module_name.lower() == parsed_module_name.lower()
+            )
+            modules_dict = {module_name: self.utils.modules_dict[module_name]}
+        elif message.clean_content.strip().endswith("test module"):
+            return Response(
+                confidence=10,
+                text="Yeah but what module?",
```

Comment:
```suggestion
                text="Yeah but which module?",
```
if you want to be grammatically precise...